File: packages/treeBurst/src/treeBurst/runtime/UnmanagedHandle.ts
import { LogMarker } from "../../prettyPrint/ObjectDescription"
import { ManagedObject } from "./ManagedObject"

export class UnmanagedHandle extends ManagedObject {
    public override[LogMarker.CUSTOM]() {
        return this
    }

    public override toString(): string {
        return this.name == null ? "<unmanaged>" : `<unmanaged ${this.name}>`
    }

    constructor(
        prototype: ManagedObject | null,
        public readonly value: any,
    ) { super(prototype) }
}

File: packages/treeBurstJava/src/main/java/bt7s7k7/treeburst/runtime/UnmanagedHandle.java
package bt7s7k7.treeburst.runtime;

public class UnmanagedHandle extends ManagedObject {

	public final Object value;

	public UnmanagedHandle(ManagedObject prototype, Object value) {
		super(prototype);
		this.value = value;
	}

	@Override
	public String toString() {
		return this.name == null ? "<unmanaged>" : String.format("<unmanaged %s>", this.name);
	}

	public Object getValue() {
		return value;
	}
}
